manifests/config.pp

```diff
@@ -50,6 +50,14 @@
     path => "${foreman_proxy::config_dir}/settings.yml",
   }
 
+  $uuid_fact_signature = foreman_proxy::hmac_signature($foreman_proxy::oauth_consumer_secret, $foreman_proxy::instance_id)
+  file { 'smart_proxy_uuid.json':
+    path    => '/opt/puppetlabs/facter/facts.d/smart_proxy_uuid.json',
+    owner   => $foreman_proxy::user,
+    mode    => '0644',
```

Comment:
That means anybody who has access to the host can read it and replicate the file (with the signature) on a different host?

Reply:
Good catch

Reply:
The signature is reported as a fact too, right? So anyone who can view_facts in Foreman sees it?

Reply:
It is imported, used to validate the "value" fact and then is discarded https://github.com/theforeman/foreman/pull/8254/commits/6a22ae032b4e322434392fc8604f157daf5e9a03